satpy/readers/viirs_sdr.py

```diff
@@ -248,25 +248,43 @@ def scale_swath_data(self, data, scaling_factors):
         data = data * factors[:, 0] + factors[:, 1]
         return data
 
-    def adjust_scaling_factors(self, factors, file_units, output_units):
-        """Adjust scaling factors."""
-        if file_units == output_units:
-            LOG.debug("File units and output units are the same (%s)", file_units)
-            return factors
-        if factors is None:
-            return None
-        factors = factors.where(factors != -999., np.float32(np.nan))
-
+    @staticmethod
+    def _scale_factors_for_units(factors, file_units, output_units):
         if file_units == "W cm-2 sr-1" and output_units == "W m-2 sr-1":
-            LOG.debug("Adjusting scaling factors to convert '%s' to '%s'", file_units, output_units)
+            LOG.debug("Adjusting scaling factors to convert '%s' to '%s'",
+                      file_units, output_units)
             factors = factors * 10000.
-            return factors
         elif file_units == "1" and output_units == "%":
-            LOG.debug("Adjusting scaling factors to convert '%s' to '%s'", file_units, output_units)
+            LOG.debug("Adjusting scaling factors to convert '%s' to '%s'",
+                      file_units, output_units)
             factors = factors * 100.
-            return factors
         else:
+            LOG.debug("Not sure how to perform unit conversion '%s' to '%s'",
+                      file_units, output_units)
```

Comment:
I made this a debug message because I wasn't actually sure when this applies. I'll double check what the "promises" are for file_units and output_units and make this an exception if necessary.

Reply:
pint? ;)

Reply:
Yeah, not throwing in another dependency yet...